modules/market/src/main/java/com/opengamma/strata/market/interpolator/InterpolatorCurveExtrapolator.java

```diff
@@ -69,46 +66,42 @@ public String toString() {
    * Bound extrapolator.
    */
   static class Bound implements BoundCurveExtrapolator {
-    private final Interpolator1D interpolator;
-    private final Interpolator1DDataBundle dataBundle;
+    private AbstractBoundCurveInterpolator interpolator;
 
     Bound(DoubleArray xValues, DoubleArray yValues, BoundCurveInterpolator interpolator) {
-      ArgChecker.isTrue(interpolator instanceof StandardBoundCurveInterpolator);
-      StandardBoundCurveInterpolator mathInterpolator = ((StandardBoundCurveInterpolator) interpolator);
-      this.dataBundle = mathInterpolator.getDataBundle();
-      this.interpolator = mathInterpolator.getInterpolator();
+      this.interpolator = (AbstractBoundCurveInterpolator) interpolator;
```

Comment:
Suggest adding `ArgChecker.isTrue(interpolator instanceof AbstractBoundCurveInterpolator);` to get a better error message